src/components/Login.tsx

```diff
@@ -1,9 +1,135 @@
- const Login:React.FC = () => {
-    return (
-        <div>
-            <p>Login</p>
-        </div>
-    )
-}
-
-export default Login;
\ No newline at end of file
+import { useRef, useState } from "react";
+import { checkValidEmail } from "./utils/validate";
+import HeaderComponent from "./ui/HeaderComponent";
+import { Link } from "react-router-dom";
+import { auth } from "./utils/firebase";
+import { createUserWithEmailAndPassword, signInWithEmailAndPassword } from "firebase/auth";
+
+const Login: React.FC = () => {
+  const [isSignUp, setIsSignUp] = useState<boolean>(false);
+  const [errorMessage, setIsErrorMessage] = useState<string | null>(null);
+
+  const name = useRef<HTMLInputElement>(null);
+  const email = useRef<HTMLInputElement>(null);
+  const password = useRef<HTMLInputElement>(null);
+
+  const signUp = isSignUp ? "Already an user!" : "New to Netflix?";
+  const signBtn = isSignUp ? "Sign Up" : "Sign In";
+  const signUpLink = isSignUp ? "Sign in now" : "Sign up now";
+  console.log("page = ", signBtn);
+  function handleLoginPage() {
+    setIsSignUp(prevVal => !prevVal);
```

Comment:
other function call might not be required, it can take !prevVal to set the State

Reply:
it can have side effects, so this is safer method